go/tendermint/apps/scheduler/scheduler.go

```diff
@@ -621,6 +642,43 @@ func (app *schedulerApplication) electValidators(ctx *abci.Context, beacon []byt
 	return nil
 }
 
+func publicKeyMapToSliceByStake(entMap map[signature.MapKey]bool, entityStake *stakeAccumulator, beacon []byte) ([]signature.PublicKey, error) {
+	// Sort, then shuffle the list of entities.
```

Comment:
We discussed this over slack already, but for the public record:
 1. Sort public keys lexographically to go from the map to a slice.
 2. Shuffle the sorted slice from 1, to avoid people trying to game the tie-breaking case by generating public keys that sort lexographically earlier.
 3. Sort by balance (reversed) to get a slice of entities by descending balance.